lcls_naming_tool/lcls_naming_tool.py

```diff
@@ -0,0 +1,309 @@
+
+        FFFFF:CCC
+        FFFFF:CCC:NN
+        FFFFF:CCC:XXXX
+        FFFFF:CCC:NN:XXXX
+        FFFFF:GGG:CCC
+        FFFFF:GGG:CCC:NN
+        FFFFF:GGG:CCC:XXXX
+        FFFFF:GGG:CCC:NN:XXXX
+
+    This script only checks the validity of PV and device names.
+'''
+
+import argparse
+import subprocess
+import sys
+import json
+import re
+
+
+fc_dict = {}
+fg_dict = {}
+ccc_dict = {}
+beam_sources = {'K': '', 'L': ''}
+beam_numbers = {'0': '', '1': '', '2': '', '3': '', '4': '', '5': ''}
+
+
+def get_version():
+    result = subprocess.run(
+        ['git', 'describe', '--tags'], 
+        capture_output=True, 
+        text=True
+    )
+    return (str(result.stdout).strip().split('-'))[0]
+
+def display_version():
+    version = get_version()
+
+    parser = argparse.ArgumentParser(description="Current version of LCLS Naming Tool")
+    parser.add_argument('-v', '--version', action='store_true')
+    args = parser.parse_args()
+    
+    if args.version:
+        print(version)
+
+
+def load_taxons():
+    global fc_dict
+    global fg_dict
+    global ccc_dict
+
+    # Read the json files containing all the taxons
+    with open('taxons/functional_component_taxon.json') as fc_file:
+        fc_dict = json.load(fc_file)
+
+    with open('taxons/fungible_element_taxon.json') as fg_file:
+        fg_dict = json.load(fg_file)
+
+    with open('taxons/ccc_taxon.json') as ccc_file:
+        ccc_dict = json.load(ccc_file)
+
+
+def functional_component_is_valid(fc_taxon):
+
+    fc_length = len(fc_taxon)
+    
+    if (4 < len(fc_taxon) < 9):
+        fc_prefix = fc_taxon[:2]  # max length is currently 2 letters
+        fc_seq_num = fc_taxon[2:fc_length-2]  # max length set to 4
+        fc_source_ltr = fc_taxon[fc_length-2:fc_length-1] # max length is currently 1 letter
+        fc_beam_num = fc_taxon[fc_length-1:fc_length]     # max length is currently 1 digit
+    else:
+        return False
+
+    # validate FFFFF
+    try:
+        fc_dict[fc_prefix]
+        beam_sources[fc_source_ltr]
+        beam_numbers[fc_beam_num]
+
+        try:
+            int(fc_seq_num)
+
+            try:
+                1 / int(fc_seq_num[0])  # zero padding not allowed
+            except 0:
+                return False
+
+        except:
+            return False
+
+    except KeyError:
+        return False
+
+    else:
+        return True
+
+
+def fungible_is_valid(fg_taxon):
+    try:
+        fg_elements = fg_taxon.replace(
+            '_', ' ').split()
+
+        for name in fg_elements:
+            fg_dict[name]
+
+    except KeyError:
+        return False
+
+    else:
+        return True
+
+
+def constituent_component_is_valid(ccc_taxon):
+    try:
+        ccc_dict[ccc_taxon]
+    except:
+        return False
+    else:
+        return True
+
+
+def increment_is_valid(nn_taxon):
+    # check the increment is an int and has at least 2 digits
+    if len(nn_taxon) >= 2:
+        try:
+            int(nn_taxon)
+        except 0:
```

Comment:
This file is actually the first time I've seen `except 0`. I'm not sure where this comes from or what it's supposed to do in Python but normally this would be something like:
```suggestion
        except Exception:
```

Reply:
I think this is actually not allowed
``` python
TypeError                                 Traceback (most recent call last)
Input In [1], in <cell line: 1>()
      1 try:
      2     int('dfsa')
----> 3 except 0:
      4     print('0')

TypeError: catching classes that do not inherit from BaseException is not allowed
```

Reply:
Hmm I found in a solution on stackoverflow. I updated this to use `except ValueError` instead.